### DISCUSSION TITLE
What is the sound packet for Lightning?

### DISCUSSION THREAD

**Discussion by Jark5455** [FabricMC/fabric-api#1764] [Mod Dev Support]
Hello, I am trying to listen to when a lightning sound packet is received by the client. Currently, I am doing this by reflecting the `Channel` from `MinecraftClient.getInstance().getNetworkHandler()` and then inserting my own channel duplex handler in the player's channel pipeline. I am listening for `PlaySoundS2CPacket`s but I am not receiving this packet when I hear lightning in game.

**Comment by warjort** [comment]
The game handles lightning as an entity (similar to a mob or experience orb).

This entity ticks on both the client and server.
It is the tick on the client that decides to play the sound so there will be no S2C sound packet.

See: ````net.minecraft.entity.LightningEntity.tick()````

I am not sure what you are are trying to achieve. But I guess the packet you want is the ````EntitySpawnS2CPacket```` with an EntityType of ````EntityType.LIGHTNING_BOLT````?

**Reply by Jark5455** [reply]
I believe is an exception to this though. If I am correct Lightning can be heard from 140,000 blocks away from any direction, it would be unreasonable to load a faraway chunk on the client to render the entity and the sound, am I right?

**Reply by warjort** [reply]
I am not sure what you are referring to?

Roughly speaking, the server keeps track of what chunks the player is interested in (the nearby ones) and only the entity events in those chunks are sent to the player.
So a player a long distance from a lightning bolt entity wouldn't even know it exists because the server wouldn't send any packets.

I might be wrong about that. There are many hard codings and special cases in the minecraft code. Maybe you have reference for what you are talking about? I don't see anything obvious in the lighting entity code.

**Reply by warjort** [reply]
The default configuration for ````EntityType.LIGHTNING_BOLT```` sets the ````maxTrackingDistance```` to 16 chunks which is 256 blocks.
This can be modified by configuration settings, but it isn't close to the 140,000 number you quote?

**Reply by Jark5455** [reply]
Source: https://minecraft.fandom.com/el/wiki/Thunderstorm#:~:text=It%20can%20be%20heard%20by,from%20the%20lightning%20strike%20itself.
Many other people have also talked about it. Also If I stand around ~2000 blocks away from spawn and summon lightning  at 0, 0 I can still hear it.

**Reply by Jark5455** [reply]
Im looking fro the packet that is sent if the lightning is out of the players chunk loading distance

**Comment by warjort** [comment]
> Also If I stand around ~2000 blocks away from spawn and summon lightning at 0, 0 I can still hear it.

What version of Minecraft are you using? When I try that on 1.17.1, I **can't** hear the lightning strike.

The link you posted is quite old. Its change log only goes up to 1.12

**Reply by Jark5455** [reply]
I am on 1.17.1 but I can hear it perfectly fine, a little faint, but I can still hear it. Are you sure you are spawning the lighning in spawn chunks?

**Reply by Jark5455** [reply]
or maybe im trippin

**Reply by Jark5455** [reply]
After a bit of research, it seems that in older versions it was sent through EntitySpawnGlobalS2CPacket, but it was recently deleted in newer versions, I'll look more into it.
